### ISSUE TITLE
SG Site Scanner Malware Detected

### ISSUE THREAD

**Issue by GWS14** [Designmodocom/WP-Maintenance-Mode#217]
Hi there

I just received a security/ malware warning from SiteGround Site Scanner, stating that malware had been detected in 3 x URLs:

https://mydomain.com.au/
https://mydomain.com.au/404testpage4525d2fdc
https://mydomain.com.au/404javascript.js

They provided the view details for each of the pages, and on each page I noted that the issue seemed to relate to the 'Bot' function in WP-Maintenance-Mode, which had just been implemented.

`http://labs.sucuri.net/db/malware/warning.maintenance?1 <!DOCTYPE html><html><head><meta charset="UTF-8"><title>Maintenance mode</title><meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" /><meta name="author" content="Solutions" /><meta name="description" content="Solutions - designing websites that simply work!" /><meta name="keywords" content="Maintenance Mode" /><meta name="robots" content="index, follow" /><link rel="stylesheet" href="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/css/style.min.css"><link rel="stylesheet" href="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/css/style.bot.min.css"><style>.wrap h1 { color: #ffffff; }.wrap h2 { color: #000000; }.background { background: url(https://mydomain.au/wp-content/uploads/2019/06/coming_soon1.jpg) no-repeat center top fixed; background-size: cover; }.wrap h3, .wrap .subscribe_wrapper { color: #ffffff; }</style></head><body class="background bot"><div class="wrap"><h1>Maintenance mode</h1></div><!-- .wrap --><div class="bot-container"> <!-- WP Bot --><div class="bot-chat-wrapper"> <!-- Chats --><div class="chat-container cf"></div> <!-- User input --><div class="input"></div> <!-- User choices --><div class="choices cf"></div></div> <!-- /WP Bot --></div><div class="bot-error"><p></p></div><div class="wrap under-bot"></div> <script type='text/javascript'>var wpmm_vars = {"ajax_url": "https://mydomain.com.au/wp-admin/admin-ajax.php"};</script> <script type='text/javascript'>var botVars = {"validationName":"Please type in your name.","validationEmail":"Please type in a valid email address.","uploadsBaseUrl":"https:\/\/goodweb.com.au\/wp-content\/uploads\/","typeName":"Type your name here\u2026","typeEmail":"Type your email here\u2026","send":"Send"}</script> <script src="https://mydomain.com.au/wp-includes/js/jquery/jquery.js"></script> <script src="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/js/scripts.min.js"></script> <script src="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/js/jquery.validate.min.js"></script> <script src="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/js/bot.mi`

I have deactivated the 'Bot' and forced a re-scan on Siteground Site Scanner, to determine whether it was just the 'Bot' activation or something else. I'm currently waiting for the scan results.

regards
GWS14

**Comment by GWS14** [comment]
Scan results returned, when just using the normal maintenance page.
Its worst than before. Now its says on pages have malware and all the pages extracts that the scanner provides show your plugin:

`http://labs.sucuri.net/db/malware/warning.maintenance?1 <!DOCTYPE html><html><head><meta charset="UTF-8"><title>Maintenance mode</title><meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" /><meta name="author" content="Good Web Solutions" /><meta name="description" content="Good Web Solutions - designing websites that simply work!" /><meta name="keywords" content="Maintenance Mode" /><meta name="robots" content="index, follow" /><link rel="stylesheet" href="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/css/style.min.css"><style>.wrap h1 { color: #ffffff; }.wrap h2 { color: #000000; }.background { background: url(https://mydomain.com.au/wp-content/uploads/2019/06/coming_soon1.jpg) no-repeat center top fixed; background-size: cover; }.wrap h3, .wrap .subscribe_wrapper { color: #ffffff; }</style></head><body class="background"><div class="wrap"><h1>Maintenance mode</h1><h2><div style="background-color: #f5f5f5;padding: 1em"><p style="font-size: 1.5em">Sorry for the inconvenience.<br />Good Web Solutions is currently undergoing scheduled maintenance.<br />Thank you for your understanding.</p></div></h2></div> <script type='text/javascript'>var wpmm_vars = {"ajax_url": "https://mydomain.com.au/wp-admin/admin-ajax.php"};</script> <script src="https://mydomain.com.au/wp-includes/js/jquery/jquery.js"></script> <script src="https://mydomain.com.au/wp-content/plugins/wp-maintenance-mode/assets/js/scripts.min.js"></script> </body></html> `

**Comment by georgejipa** [comment]
@GWS14 Thanks for telling us. 

The code of the plugin is public, and anyone can see there is no malware. So, I am going to contact Siteground to ask what happened.

**Comment by georgejipa** [comment]
I've contacted them, and now we have to wait for an answer.

Until that, I think it is a false positive because of the description of the warning (sucuri.net provides the functionality for Siteground Site Scanner):
http://labs.sucuri.net/db/malware/warning.maintenance?1
 
> Description: Web page cannot be reliably scanned as it says that it is in the Maintenance Mode.
> 
> **Most likely website undergoes some changes at the moment and you should scan it later when the maintenance is over.** It also can be a sign of a security problem when hackers corrupted the site and the "maintenance" page is used as a fall back.

**Comment by GWS14** [comment]
Thanks for quick responses George. 
Have replaced with another maintenance plugin and now scan says all is clean.

**Comment by georgejipa** [comment]
Okay, it's your choice. I will keep in touch with Siteground anyway because we want to solve the issue.
